ownCloud/Client/Sharing/GroupSharingTableViewController.swift

```diff
@@ -0,0 +1,465 @@
+		}
+	}
+
+	func removeShareSections() {
+		OnMainThread {
+			let types : [OCShareType] = [.userShare, .groupShare, .remote]
+			for type in types {
+				let identifier = "share-section-\(String(type.rawValue))"
+				if let section = self.sectionForIdentifier(identifier) {
+					self.removeSection(section)
+				}
+			}
+
+			let identifier = "action-section"
+			if let section = self.sectionForIdentifier(identifier) {
+				self.removeSection(section)
+			}
+		}
+	}
+
+	func resetTable(showShares : Bool) {
+		removeShareSections()
+		messageView?.message(show: false)
+		if let section = self.sectionForIdentifier("search-results") {
+			self.removeSection(section)
+		}
+		if shares.count > 0 && showShares {
+			self.addShareSections()
+		}
+	}
+
+	// MARK: - Sharing Helper
+
+	func share(at indexPath : IndexPath) -> OCShare? {
+		var type : OCShareType?
+		switch indexPath.section {
+		case 0:
+			type = .userShare
+		case 1:
+			type = .groupShare
+		case 2:
+			type = .remote
+		default:
+			break
+		}
+
+		if type != nil {
+			let shares = self.shares.filter { (OCShare) -> Bool in
+				if OCShare.type == type {
+					return true
+				}
+				return false
+			}
+			if shares.indices.contains(indexPath.row) {
+				return shares[indexPath.row]
+			}
+		}
+
+		return nil
+	}
+
+	func canEdit(share: OCShare) -> Bool {
+		if core?.connection.loggedInUser?.userName == share.owner?.userName || core?.connection.loggedInUser?.userName == share.itemOwner?.userName {
+			return true
+		}
+
+		return false
+	}
+
+	// MARK: - UISearchResultsUpdating Delegate
+
+	func updateSearchResults(for searchController: UISearchController) {
+		guard let text = searchController.searchBar.text else { return }
+		if text.count > core?.connection.capabilities?.sharingSearchMinLength?.intValue ?? 1 {
+			recipientSearchController?.searchTerm = text
+		} else if searchController.isActive {
+			resetTable(showShares: false)
+		}
+	}
+
+	func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+		self.resetTable(showShares: true)
+		self.searchController?.searchBar.isLoading = false
+	}
+
+	func searchControllerHasNewResults(_ searchController: OCRecipientSearchController, error: Error?) {
+		OnMainThread {
+			guard let recipients = searchController.recipients else {
+				self.messageView?.message(show: true, imageName: "icon-search", title: "No matches".localized, message: "There is no results for this search".localized)
+				return
+			}
+
+			self.messageView?.message(show: false)
+			var rows : [StaticTableViewRow] = []
+			for recipient in recipients {
+
+				guard let itemPath = self.item.path else { continue }
+				var title = ""
+				if recipient.type == .user {
+					guard let displayName = recipient.displayName else { continue }
+					title = displayName
+				} else {
+					guard let displayName = recipient.displayName else { continue }
+					let groupTitle = "(Group)".localized
+					title = "\(displayName) \(groupTitle)"
+				}
+
```

Comment:
From the department of nice-to-have-but-not-required: would be awesome an icon representing a `user` or `group` would be in front of each line in the search results.

Reply:
@felix-schwarz I have implemented the `OCUser.avatar` but this is always nil. I set the profile picture in the settings for the user.